packages/react-i18n-webpack-plugin/src/react-i18n-webpack-plugin.ts

```diff
@@ -0,0 +1,264 @@
+
+          // replace useI18n & withI18n call arguments
+          parser.hooks.evaluate
+            .for('CallExpression')
+            .tap(PLUGIN_NAME, (originalExpression: CallExpression) => {
+              if (
+                parser.state.module.resource.indexOf('node_modules') !== -1 &&
+                !parser.state.module.resource.endsWith('tsx') &&
+                !parser.state.i18nImports
+              ) {
+                return;
+              }
+
+              const componentPath = parser.state.module.resource;
+              const componentDir = parser.state.module.context;
+              const identifierName = parser.state.i18nImports.get(
+                componentPath,
+              );
+
+              if (
+                !identifierName ||
+                originalExpression.callee.type !== 'Identifier'
+              ) {
+                return;
+              }
+
+              let expression: CallExpression | null = null;
+              if (originalExpression.callee.name === 'compose') {
+                const foundExpressions: Array<
+                  Expression | SpreadElement
+                > = originalExpression.arguments.filter(
+                  node =>
+                    node.type === 'CallExpression' &&
+                    node.callee.type === 'Identifier' &&
+                    node.callee.name === identifierName,
+                );
+
+                if (foundExpressions.length > 0) {
+                  expression = foundExpressions[0] as CallExpression;
+                }
+              } else if (originalExpression.callee.name === identifierName) {
+                expression = originalExpression;
+              }
+
+              // skip calls where consumer manually added arguments
+              if (!expression || expression.arguments.length > 0) {
+                return;
+              }
+
+              const translationFiles = getTranslationFiles(parser);
+              if (translationFiles.length === 0) {
+                return;
+              }
+
+              // check if fall back exist
+              const fallBackExist = translationFiles.find(
+                translationFile =>
+                  translationFile === `${this.options.fallbackLocale}.json`,
+              );
+
+              const fallBackFileRelativePath = path.join(
+                './',
+                TRANSLATION_DIRECTORY_NAME,
+                `${this.options.fallbackLocale}.json`,
+              );
+
+              if (!fallBackExist) {
+                compilation.errors.push(
+                  `${componentPath}\n` +
+                    `${identifierName} 's arguments was not automatically filled in because` +
+                    `fallback translation file was not found at ${fallBackFileRelativePath} \n`,
+                );
+                return;
+              }
+
+              // Add a top-level fallbackLocale import
+              const fallbackLocaleID = camelCase(this.options.fallbackLocale);
```

Comment:
Hmm I feel like we need a "safer" name; this looks like it will just be `en`, which has a non-zero chance of conflicting with an actual variable in scope.

Reply:
What about `en-US`? If I am making `fallbackLocale` a option it may not be `en` in the future

Reply:
sure, `enUs` is not safe either. Any name that is just camelcasing could conflict with the app code. The usual approach is to use a variable name that is very unlikely to conflict (one with random characters at the end, double underscore prefix, etc)

Reply:
aw, I thought the problem was the `camelCase`, I see what you mean now.